bindings/node/lib/mongocryptdManager.js

```diff
@@ -0,0 +1,123 @@
+'use strict';
+
+const spawn = require('child_process').spawn;
+const readFile = require('fs').readFile;
+
+/**
+ * @typedef AutoEncryptionExtraOptions
+ * @prop {string} [mongocryptdURI] overrides the uri used to connect to mongocryptd
+ * @prop {boolean} [mongocryptdBypassSpawn=false] if true, autoEncryption will not spawn a mongocryptd
+ * @prop {string} [mongocryptdSpawnPath] the path to the mongocryptd executable
+ * @prop {string[]} [mongocryptdURI] command line arguments to pass to the mongocryptd executable
+ */
+
+module.exports = function() {
+  const platform = require('os').platform;
+
+  const MONGOCRYPTD_PID_FILE = 'mongocryptd.pid';
+  const STATE = {
+    NO_PID_FILE: Symbol('NO_PID_FILE'),
+    EMPTY_PID_FILE: Symbol('EMPTY_PID_FILE'),
+    VALID_PID_FILE: Symbol('VALID_PID_FILE')
+  };
+
+  const checkIntervalMS = 100;
```

Comment:
this seems a bit fast, how about 500ms?

Reply:
I was going off of the example [here](https://github.com/markbenvenuto/mongo/blob/cryptd_json/mongocryptd_controller.py#L65). tbh, I'd like this to be spec-d, but I'm fine with 500ms.

Reply:
this has the effect of making every autoEncryption startup take at least 500ms, which among other things significantly slows down the test runner.

Could we perhaps do 50 ms intervals with 20 attempts? that way, if it does indeed take a full second to start up, we will catch it, but we will immediately resolve if we have it?